### DISCUSSION TITLE
QMK pre process record user

### DISCUSSION THREAD

**Discussion by filterpaper** [getreuer/qmk-keymap#35] [Ideas]
A new pre process record "shim" was merged in https://github.com/qmk/qmk_firmware/pull/20584. It adds a new `pre_process_record_user` function for user space that will be called before quantum processing. This might be useful for handling mod tap keys before they are sent to action tapping, instead of "post processing" them like "Achordion".

I use it to capture the tap record that follows a mod tap undergoing resolution in action tapping interval. That tap record is then used for the current mod tap decision based on its unilateral or bilateral context.

**Comment by getreuer** [comment]
This is great! Yes, a user hook at this point is exactly what's needed for Achordion. Thanks so much for developing this and for the heads up!

**Reply by drashna** [reply]
Messed with it, a bit. It's definitely not a straight replacement, and I wouldn't be surprised if a decent chunk of the code would need to be re-written.  But would likely be very worth it.
